src/AppBundle/Model/User/Middleware/NameSuggestionMiddleware.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of the Sententiaregum project.
+ *
+ * (c) Maximilian Bosch <maximilian.bosch.27@gmail.com>
+ * (c) Ben Bieler <benjaminbieler2014@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace AppBundle\Model\User\Middleware;
+
+use AppBundle\Model\User\DTO\CreateUserDTO;
+use AppBundle\Model\User\Util\NameSuggestion\Suggestor\SuggestorInterface;
+use AppBundle\Validator\Constraints\UniqueProperty;
+use SimpleBus\Message\Bus\Middleware\MessageBusMiddleware;
+use Symfony\Component\Validator\ConstraintViolationListInterface;
+
+/**
+ * Middleware which is responsible for the name suggestion generation if the DTO validation failed.
+ *
+ * @author Maximilian Bosch <maximilian.bosch.27@gmail.com>
+ */
+class NameSuggestionMiddleware implements MessageBusMiddleware
+{
+    const SUGGESTIONS = 'suggestions';
+
+    /**
+     * @var SuggestorInterface
+     */
+    private $suggestor;
+
+    /**
+     * Constructor.
+     *
+     * @param SuggestorInterface $suggestor
+     */
+    public function __construct(SuggestorInterface $suggestor)
+    {
+        $this->suggestor = $suggestor;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle($message, callable $next)
+    {
+        if (!$message instanceof CreateUserDTO
+            || (!$info = $message->getInfo())
```

Comment:
second condition is unnecessary actually. If this message is not a `CreateUserDTO` it's not validatable. And validatable DTOs have this info. Basically that's a hack to simplify tests and keep them green.